docs/source/examples/Widget Custom.ipynb

```diff
@@ -0,0 +1,785 @@
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### sync=True traitlets"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Traitlets is an IPython library for defining type-safe properties on configurable objects. For this tutorial you do not need to worry about the *configurable* piece of the traitlets machinery. The `sync=True` keyword argument tells the widget framework to handle synchronizing that value to the browser. Without `sync=True`, attributes of the widget won't be synchronized with the front-end.\n",
+    "\n",
+    "<div class=\"alert alert-info\">\n",
+    "Syncing mutable types\n",
+    "    \n",
+    "Please keep in mind that mutable types will not necessarily be synced when they are modified. For example appending an element to a `list` will not cause the changes to sync. Instead a new list must be created and assigned to the trait for the changes to be synced.\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Other traitlet types"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Unicode, used for `_view_name`, is not the only Traitlet type, there are many more some of which are listed below:  \n",
+    "\n",
+    "- Any\n",
+    "- Bool\n",
+    "- Bytes\n",
+    "- CBool\n",
+    "- CBytes\n",
+    "- CComplex\n",
+    "- CFloat\n",
+    "- CInt\n",
+    "- CLong\n",
+    "- CRegExp\n",
+    "- CUnicode\n",
+    "- CaselessStrEnum\n",
+    "- Complex\n",
+    "- Dict\n",
+    "- DottedObjectName\n",
+    "- Enum\n",
+    "- Float\n",
+    "- FunctionType\n",
+    "- Instance\n",
+    "- InstanceType\n",
+    "- Int\n",
+    "- List\n",
+    "- Long\n",
+    "- Set\n",
+    "- TCPAddress\n",
+    "- Tuple\n",
+    "- Type\n",
+    "- Unicode\n",
+    "- Union\n",
+    "\n",
+    "\n",
+    "Not all of these traitlets can be synchronized across the network, only the JSON-able traits and Widget instances will be synchronized."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Front end (TypeScript)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Models and views"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The IPython widget framework front end relies heavily on [Backbone.js](http://backbonejs.org/).  Backbone.js is an MVC (model view controller) framework.  Widgets defined in the back end are automatically synchronized with generic Backbone.js models in the front end.  The traitlets are added to the front end instance automatically on first state push.  The `_view_name` trait that you defined earlier is used by the widget framework to create the corresponding Backbone.js view and link that view to the model."
```

Comment:
Perhaps add some more explanations:

```suggestion
    "The IPython widget framework front end relies heavily on [Backbone.js](http://backbonejs.org/).  Backbone.js is an MVC (model view controller) framework.  Widgets defined in the back end are automatically synchronized with Backbone.js `Model` in the front end. Each front end `Model` handles the widget data and state, and can have any number of associate `View`s. In the context of a widget the `Views` are what render objects for the user to interact with, and the Model handles communication with the Python objects.\n",
    "\n",
    "On the first state push from python the synced traitlets are added automatically. The `_view_name` trait that you defined earlier is used by the widget framework to create the corresponding Backbone.js view and link that view to the model.\n"
```


May also be worthwhile to discuss the defaults. Though I'm not actually 100% how/when they are important. This may come up for a reader especially in the context of `_view_name` where the text claims it is set by python, but the code seems to imply that it also comes from `defaults`